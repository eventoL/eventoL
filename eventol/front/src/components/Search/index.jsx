import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';

export default class Search extends React.PureComponent {
  static propTypes = {
    onChange: PropTypes.func,
    onEnter: PropTypes.func.isRequired,
  };

  static defaultProps = {
    onChange: () => {},
  };

  state = {
    searchTerm: '',
  };

  handleKeyUp = ({key}) => {
    if (key === 'Enter') {
      const {onEnter} = this.props;
      const {searchTerm} = this.state;
      onEnter(searchTerm);
    }
  };

  handleChange = ({target: {value: searchTerm}}) => {
    this.setState({searchTerm});
    const {onChange} = this.props;
    onChange(searchTerm);
  };

  render() {
    const {searchTerm} = this.state;
    return (
      <div className="search" id="search">
        <input
          id="search-input"
          onChange={this.handleChange}
          onKeyUp={this.handleKeyUp}
          placeholder={gettext('Search by title...')}
          type="search"
          value={searchTerm}
        />
      </div>
    );
  }
}
