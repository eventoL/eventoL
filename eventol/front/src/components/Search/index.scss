@import '../../styles/variables';
@import '../../styles/commons';

.search {
  @extend %flex-element;
  @extend %center-element;
  left: 0;
  pointer-events: none;
  position: absolute;
  width: 100%;

  input {
    -moz-appearance: none;
    -webkit-appearance: none;
    -webkit-transition: border .25s ease, width .125s ease .125s;
    appearance: none;
    background: transparent;
    border: 2px solid $black;
    border-radius: 37px;
    color: $text-color;
    font-family: 'Lato', sans-serif;
    font-size: 14px;
    font-weight: 300;
    height: 37px;
    outline: none;
    padding: 0 10px;
    pointer-events: all;
    transition: border .25s ease, width .125s ease .125s;
    width: 35vw;
    min-width: 150px;
  }

  input:focus {
    border: 2px solid $eventol-color;
    width: 45vw;
  }
}
