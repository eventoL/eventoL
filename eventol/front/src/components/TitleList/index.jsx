import _ from 'lodash';
import React, {useState, useEffect, useCallback} from 'react';
import PropTypes from 'prop-types';

import SliderItems from '../SliderItems';
import {getUrl} from '../../utils/api';
import {getApiFullUrl} from '../../utils/urls';
import {parseEventToItem, emptyEventItem} from '../../utils/events';

import './index.scss';

const TitleList = props => {
  const [data, setData] = useState({results: []});
  const [mounted, setMounted] = useState(false);
  const {title, id, showEmpty, url} = props;

  const loadContent = useCallback(() => {
    const fullUrl = getApiFullUrl(url);
    getUrl(fullUrl).then(setData);
  }, [url, setData]);

  useEffect(() => {
    if (!_.isEmpty(url)) {
      setMounted(true);
      loadContent();
    }
  }, [loadContent, setMounted, url]);

  const {results} = data;
  let itemsData = '';
  if (results) {
    itemsData = results.map(parseEventToItem);
  }
  if (_.isEmpty(itemsData)) {
    if (!showEmpty) return null;
    return (
      <div className="title-list" data-loaded={mounted} id={id}>
        <div className="category-title">
          <h1>{title}</h1>
          <SliderItems itemsData={[emptyEventItem]} sliderId={`${id}_empty`} />
        </div>
      </div>
    );
  }
  return (
    <div className="title-list" data-loaded={mounted} id={id}>
      <div className="category-title">
        <h1>{title}</h1>
        <SliderItems itemsData={itemsData} sliderId={id} />
      </div>
    </div>
  );
};

TitleList.propTypes = {
  id: PropTypes.string.isRequired,
  showEmpty: PropTypes.bool,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

TitleList.defaultProps = {
  showEmpty: false,
};

export default TitleList;
