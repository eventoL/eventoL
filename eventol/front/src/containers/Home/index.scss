@import '../../styles/index';

html,
body,
div,
span,
p,
a,
ul,
li,
header,
nav {
  border: 0;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

ol,
ul {
  list-style: none;
}

q,
blockquote {
  quotes: none;

  &::before,
  &::after {
    content: '';
  }
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}

body {
  background: $body-background;
  color: $body-color;
  font-family: 'Lato', sans-serif;
  line-height: 1;
}

h2 {
  font-size: 24px;
  font-weight: 600;
  line-height: 1.4;
  margin-bottom: .5em;
  margin-top: 5px;
}

p {
  font-size: 15px;
  font-weight: 300;
  line-height: 1.6;
  margin-bottom: 1em;
}
