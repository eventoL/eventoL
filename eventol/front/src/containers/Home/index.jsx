import React from 'react';
import PropTypes from 'prop-types';
import withSizes from 'react-sizes';

import Hero from '../../components/Hero';
import Header from '../../components/Header';
import Search from '../../components/Search';
import TitleList from '../../components/TitleList';
import {mapSizesToProps} from '../../utils/dom';
import {
  BACKGROUND_DEFAULT,
  LOGO_HEADER_DEFAULT,
  LOGO_LANDING_DEFAULT,
} from '../../utils/constants';
import {
  getSearchUrl,
  getMyEventsUrl,
  getUpcomingEventsUrl,
  getFinishedEventsUrl,
  getEventsWithConfirmedScheduleUrl,
  getFeaturedEventsUrl,
  getRecentEventsUrl,
} from '../../utils/urls';

import './index.scss';

class Home extends React.PureComponent {
  static propTypes = {
    background: PropTypes.string,
    eventolMessage: PropTypes.string,
    handleOnChangeLanguage: PropTypes.func,
    isMobile: PropTypes.bool.isRequired,
    languages: PropTypes.arrayOf(
      PropTypes.shape({
        code: PropTypes.string,
        name: PropTypes.string,
      })
    ),
    logoHeader: PropTypes.string,
    logoLanding: PropTypes.string,
    user: PropTypes.shape({
      first_name: PropTypes.string,
      last_name: PropTypes.string,
    }),
  };

  static defaultProps = {
    background: BACKGROUND_DEFAULT,
    eventolMessage: null,
    handleOnChangeLanguage: null,
    languages: [],
    logoHeader: LOGO_HEADER_DEFAULT,
    logoLanding: LOGO_LANDING_DEFAULT,
    user: null,
  };

  state = {
    searchUrl: '',
    searched: false,
  };

  handleOnEnter = searchTerm => {
    if (searchTerm !== '') {
      const searchUrl = getSearchUrl(searchTerm);
      this.setState({searchUrl, searched: true});
    }
  };

  handlerOnChangeLanguage = languageCode => {
    const {handleOnChangeLanguage} = this.props;
    if (handleOnChangeLanguage) {
      handleOnChangeLanguage(languageCode);
    }
  };

  render() {
    const {searched, searchUrl} = this.state;
    const {
      user,
      eventolMessage,
      background,
      logoHeader,
      logoLanding,
      isMobile,
      languages,
    } = this.props;
    return (
      <div>
        <Header
          handlerOnChangeLanguage={this.handlerOnChangeLanguage}
          isMobile={isMobile}
          languages={languages}
          logoHeader={logoHeader}
          user={user}
        />
        <Hero
          background={background}
          isLogged={user !== null}
          logoLanding={logoLanding}
          message={eventolMessage}
        >
          <Search onEnter={this.handleOnEnter} />
        </Hero>
        {searched && (
          <TitleList
            id="search_results"
            showEmpty
            title={gettext('Search results')}
            url={searchUrl}
          />
        )}
        <TitleList
          id="my_events"
          title={gettext('My Events')}
          url={getMyEventsUrl()}
        />
        <TitleList
          id="recent"
          title={gettext('Recent Events')}
          url={getRecentEventsUrl()}
        />
        <TitleList
          id="featured"
          title={gettext('Featured Events')}
          url={getFeaturedEventsUrl()}
        />
        <TitleList
          id="next"
          title={gettext('Upcoming Events')}
          url={getUpcomingEventsUrl()}
        />
        <TitleList
          id="schedule_confirmed"
          title={gettext('Events with Confirmed Schedule')}
          url={getEventsWithConfirmedScheduleUrl()}
        />
        <TitleList
          id="finished"
          title={gettext('Finished Events')}
          url={getFinishedEventsUrl()}
        />
      </div>
    );
  }
}

export default withSizes(mapSizesToProps)(Home);
