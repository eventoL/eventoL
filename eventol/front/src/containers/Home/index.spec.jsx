import wait from 'waait';
import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../../components/Hero', () => 'Hero');
jest.mock('../../components/Header', () => 'Header');
jest.mock('../../components/Search', () => 'Search');
jest.mock('../../components/TitleList', () => 'TitleList');

import Search from '../../components/Search';
import TitleList from '../../components/TitleList';

import Home from '.';

describe('Home', () => {
  let tree;
  let user;
  let props;
  let element;
  let instance;
  let component;
  let background;
  let logoHeader;
  let logoLanding;
  let eventolMessage;

  const getComponent = allProps => {
    element = <Home {...allProps} />;
    component = renderer.create(element);
    instance = component.root;
    return component;
  };

  beforeEach(() => {
    background = 'background';
    logoHeader = 'logoHeader';
    logoLanding = 'logoLanding';
    eventolMessage = 'eventolMessage';
    user = {
      first_name: 'first_name',
      last_name: 'last_name',
    };
    props = {
      user,
      background,
      logoHeader,
      logoLanding,
      eventolMessage,
    };
  });

  describe('Search', () => {
    beforeEach(() => {
      component = getComponent({});
      tree = component.toJSON();
    });

    test('Should show a new TitleList for search', async () => {
      expect(tree).toMatchSnapshot();

      const value = 'searchTerm';
      instance.findByType(Search).props.onEnter(value);
      component.update(element);
      await wait(0);

      tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      const titleLists = instance.findAllByType(TitleList);
      expect(titleLists[0].props.id).toEqual('search_results');
    });

    test('Should not show a new TitleList for search with value is ""', async () => {
      expect(tree).toMatchSnapshot();

      const value = '';
      instance.findByType(Search).props.onEnter(value);
      component.update(element);
      await wait(0);

      tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      const titleLists = instance.findAllByType(TitleList);
      expect(titleLists[0].props.id).not.toEqual('search_results');
    });
  });

  describe('With user', () => {
    describe('Desktop', () => {
      describe('Default', () => {
        beforeEach(() => {
          component = getComponent({user, isMobile: false});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });

      describe('With custom data', () => {
        beforeEach(() => {
          component = getComponent({...props, isMobile: false});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });
    });

    describe('Mobile', () => {
      describe('Default', () => {
        beforeEach(() => {
          component = getComponent({user, isMobile: true});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });

      describe('With custom data', () => {
        beforeEach(() => {
          component = getComponent({...props, isMobile: true});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });
    });
  });

  describe('Without user', () => {
    describe('Desktop', () => {
      describe('Default', () => {
        beforeEach(() => {
          component = getComponent({isMobile: false});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });

      describe('With custom data', () => {
        beforeEach(() => {
          props = {
            eventolMessage,
            background,
            logoHeader,
            logoLanding,
          };
          component = getComponent({...props, isMobile: false});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });
    });

    describe('Mobile', () => {
      describe('Default', () => {
        beforeEach(() => {
          component = getComponent({isMobile: true});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });

      describe('With custom data', () => {
        beforeEach(() => {
          props = {
            eventolMessage,
            background,
            logoHeader,
            logoLanding,
          };
          component = getComponent({...props, isMobile: true});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });
    });
  });
});
