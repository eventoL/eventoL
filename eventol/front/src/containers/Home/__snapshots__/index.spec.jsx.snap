// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home With languages Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={
      Array [
        Object {
          "code": "es",
          "name": "Spanish",
        },
        Object {
          "code": "en",
          "name": "English",
        },
      ]
    }
    logoHeader="/static/manager/img/eventol-white.png"
    user={null}
  />
  <Hero
    background="/static/manager/img/background.png"
    isLogged={false}
    logoLanding="/static/manager/img/logo.png"
    message={null}
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home With user Desktop Default Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={Array []}
    logoHeader="/static/manager/img/eventol-white.png"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="/static/manager/img/background.png"
    isLogged={true}
    logoLanding="/static/manager/img/logo.png"
    message={null}
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home With user Desktop With custom data Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={Array []}
    logoHeader="logoHeader"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="background"
    isLogged={true}
    logoLanding="logoLanding"
    message="eventolMessage"
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home With user Mobile Default Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={true}
    languages={Array []}
    logoHeader="/static/manager/img/eventol-white.png"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="/static/manager/img/background.png"
    isLogged={true}
    logoLanding="/static/manager/img/logo.png"
    message={null}
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home With user Mobile With custom data Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={true}
    languages={Array []}
    logoHeader="logoHeader"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="background"
    isLogged={true}
    logoLanding="logoLanding"
    message="eventolMessage"
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home Without user Desktop Default Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={Array []}
    logoHeader="/static/manager/img/eventol-white.png"
    user={null}
  />
  <Hero
    background="/static/manager/img/background.png"
    isLogged={false}
    logoLanding="/static/manager/img/logo.png"
    message={null}
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home Without user Desktop With custom data Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={Array []}
    logoHeader="logoHeader"
    user={null}
  />
  <Hero
    background="background"
    isLogged={false}
    logoLanding="logoLanding"
    message="eventolMessage"
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home Without user Mobile Default Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={true}
    languages={Array []}
    logoHeader="/static/manager/img/eventol-white.png"
    user={null}
  />
  <Hero
    background="/static/manager/img/background.png"
    isLogged={false}
    logoLanding="/static/manager/img/logo.png"
    message={null}
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home Without user Mobile With custom data Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={true}
    languages={Array []}
    logoHeader="logoHeader"
    user={null}
  />
  <Hero
    background="background"
    isLogged={false}
    logoLanding="logoLanding"
    message="eventolMessage"
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home handleOnEnter Should not show a new TitleList for search with value is "" 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={Array []}
    logoHeader="/static/manager/img/eventol-white.png"
    user={null}
  />
  <Hero
    background="/static/manager/img/background.png"
    isLogged={false}
    logoLanding="/static/manager/img/logo.png"
    message={null}
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home handleOnEnter Should not show a new TitleList for search with value is "" 2`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={Array []}
    logoHeader="/static/manager/img/eventol-white.png"
    user={null}
  />
  <Hero
    background="/static/manager/img/background.png"
    isLogged={false}
    logoLanding="/static/manager/img/logo.png"
    message={null}
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home handleOnEnter Should show a new TitleList for search 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={Array []}
    logoHeader="/static/manager/img/eventol-white.png"
    user={null}
  />
  <Hero
    background="/static/manager/img/background.png"
    isLogged={false}
    logoLanding="/static/manager/img/logo.png"
    message={null}
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;

exports[`Home handleOnEnter Should show a new TitleList for search 2`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={Array []}
    logoHeader="/static/manager/img/eventol-white.png"
    user={null}
  />
  <Hero
    background="/static/manager/img/background.png"
    isLogged={false}
    logoLanding="/static/manager/img/logo.png"
    message={null}
  >
    <Search
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="search_results"
    showEmpty={true}
    title="Search results"
    url="?search=searchTerm&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="recent"
    title="Recent Events"
    url="?ordering=-created_at&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="featured"
    title="Featured Events"
    url="?ordering=-attendees_count&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="schedule_confirmed"
    title="Events with Confirmed Schedule"
    url="?schedule_confirmed=true&registration_is_open=true&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags"
  />
</div>
`;
